sdk2ecr

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+set -e -o pipefail
+shopt -qs failglob
+
+for opt in "$@"; do
+   optarg="$(expr "${opt}" : '[^=]*=\(.*\)')"
+   case "${opt}" in
+      --version=*) VERSION="${optarg}" ;;
+      --registry=*) REGISTRY="${optarg}" ;;
+      --sdk-name=*) SDK_NAME="${optarg}" ;;
+   esac
+done
+
+if [ -z "${VERSION}" ]; then
+  VERSION=$(make print-version | tail -1)
+fi
+
+for required_arg in VERSION REGISTRY SDK_NAME; do
+  [ -z "${!required_arg}" ] && echo "${required_arg} not set!" >&2 && exit 1
+done
+
+update_image() {
+  set -u
+
+  local local_image="${1:?}"
+  local remote_image="${2:?}"
+  local arch="${3:?}"
+
+  local host_arch
+  host_arch="$(uname -m)"
+
+  local docker_arch
+  case "${host_arch}" in
+    x86_64) docker_arch="amd64" ;;
+    aarch64) docker_arch="arm64" ;;
+    *) echo "unknown host arch ${host_arch}" >&2 && exit 1 ;;
+  esac
+
+  local alt_arch
+  case "${host_arch}" in
+    x86_64) alt_arch="aarch64" ;;
+    aarch64) alt_arch="x86_64" ;;
+    *) echo "unknown host arch ${host_arch}" >&2 && exit 1 ;;
+  esac
+
+  local alt_docker_arch
+  case "${docker_arch}" in
+    amd64) alt_docker_arch="arm64" ;;
+    arm64) alt_docker_arch="amd64" ;;
+  esac
+
+  if ! docker image inspect "${local_image}-${host_arch}" >/dev/null 2>&1 ; then
+    echo "did not find ${local_image}-${host_arch}, skipping it ..." >&2
+    return
+  fi
+
+  if ! docker tag "${local_image}-${host_arch}" "${remote_image}-${host_arch}" ; then
+    echo "failed to tag ${remote_image}-${host_arch}" >&2 && exit 1
+  fi
+
+  # push the image, if it exists locally
+  if ! docker push "${remote_image}-${host_arch}" ; then
+    echo "failed to push ${remote_image}-${host_arch}" >&2 && exit 1
+  fi
+
+  # clean up any cached local copy of the manifest
+  find "${HOME}/.docker/manifests" -type f -delete
+  find "${HOME}/.docker/manifests" -mindepth 1 -type d -delete
+
+  if ! docker manifest inspect "${remote_image}-${alt_arch}" >/dev/null 2>&1 ; then
+    echo "could not find ${remote_image}-${alt_arch}, skipping manifest creation" >&2
+    return
+  fi
+
+  local image="${remote_image#*/}"
+  local version="${image##*:}"
+  local repo="${image%:*}"
+
+  aws ecr batch-delete-image \
+    --repository-name "${repo}" \
+    --image-id imageTag="${version}"
```

Comment:
I have this change in my local copy of the script; I hit some weird errors otherwise even when the image existed and should have been deleted:

```suggestion
  aws ecr batch-delete-image \
    --repository-name "${repo}" \
    --image-id imageTag="${version}" \
    --region us-east-1
```

We might want to suppress this output since the error messages are a little confusing.

Reply:
Rather than suppressing, the errors were mitigated.